sstables/mx/bsearch_clustered_cursor.hh

```diff
@@ -268,7 +269,7 @@ public:
             uint64_t promoted_index_size,
             metrics& m,
             reader_permit permit,
-            column_values_fixed_lengths cvfl,
+            column_translation ctr,
```

Comment:
What's the motivation behind storing the entire `column_translation` object inside the cursor, instead of only `column_values_fixed_lengths`? Seems to me this refactor is extra on top of the optimization, not strictly necessary.

Reply:
ctr is lw_shared_ptr, so cheaper to copy than column_values_fixed_lengths (vector).

Reply:
You could maintain a reference to it. Since it's stable, there's no need to copy-on-write it. The sstable could maintain a stable copy.

Reply:
it's per schema version used by the read, so not stable.

Reply:
next call to sstable::get_column_translation(const schema&) may invalidate column_translation::state obtained earlier.

Reply:
I meant, sstable could compute only the primary key aspect which you're interested in, and that doesn't change.

Reply:
It could indeed. But not worth the effort here? storing shared ptr is enough of improvement.